subsys/bluetooth/host/hci_core.c

```diff
@@ -116,6 +116,9 @@ struct cmd_data {
 
 	/** Used by bt_hci_cmd_send_sync. */
 	struct k_sem *sync;
+
+	/** Response buffer. */
+	struct net_buf *rsp;
```

Comment:
While we are on the topic of renaming: Isn't it a bit weird that we now do `cmd(buf)->rsp` to get the response, which is a separate object than the command?

The struct here is called `cmd_data` which does fit the use case, but it does seem a bit odd that `cmd(buf)` returns a `cmd_data` instead of a command, and arguably the macro should have been `cmd_data(buf)`.

Reply:
yeah. you are right. `cmd_data` more suitable, but i think this change beyond this PR, maybe add separate commit for this.